### PR TITLE
fix(template): Syntax error caused by runtime template

### PR DESCRIPTION
**Pull request by lcy930323**
The motivation of this PR is to fix syntax errors caused by MissingModuleTemplate:

https://github.com/webpack/webpack/blob/master/declarations.d.ts#L10-L17

A: (function(){ }())
B: !(function(){ }())
A and B are the same effect, A don't cause syntax errors

issue:
https://github.com/webpack/webpack/issues/9187
https://github.com/webpack/webpack/issues/6191
https://github.com/webpack/webpack/issues/8762

What kind of change does this PR introduce?

refactor

Did you add tests for your changes?

Yes

Does this PR introduce a breaking change?

No

What needs to be documented once your changes are merged?

I don't think this should require any doc updates

### CODE REVIEW

**Comment by sokra** [comment]
> A: (function(){ }())
> B: !(function(){ }())
> A and B are the same effect, A don't cause syntax errors

That's not 100% true. B causes an ASI and A not. So if the previous line has no semicolon at the end, A is treated as call and B not.

``` js
// A
if(false) a
(function() { }())
// => if(false) a(function() { }())

// B
if(false) a
!(function() { }())
// => if(false) a;
// => !(function() { }())
```

Would `Object(function() { }())` work? That's safe from ASI side. It should also work with `new Object(function() { }())`

**Comment by lcy930323** [comment]
> Would Object(function() { }()) work? That's safe from ASI side. It should also work with new Object(function() { }())

yes, `Object(function() { }())`  is work ok!

**Comment by lcy930323** [comment]
TestCase general configuration cannot be replicated, so a new `RuntimeTemplateCheck.test.js` is added

File: test/RuntimeTemplateCheck.test.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+"use strict";
```
Comment (sokra) [inline test/RuntimeTemplateCheck.test.js]: This is too much code for this test case.

Please use the existing test cases system by putting your test case into `test/cases/...`

**Comment by lcy930323** [comment]
update, sse existing test cases

File: test/RuntimeTemplateCheck.test.js (hunk 1 above)
Comment (lcy930323) [reply test/RuntimeTemplateCheck.test.js]: update, use existing test cases system

**Comment by sokra** [comment]
Thanks

**Pull request merged**